Tests/SonarScanner.MSBuild.PreProcessor.Test/Infrastructure/MockObjectFactory.cs

```diff
@@ -19,46 +19,51 @@
  */
 
 using FluentAssertions;
+using Moq;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.Common.TFS;
+using SonarScanner.MSBuild.PreProcessor.Roslyn.Model;
+using TestUtilities;
 
 namespace SonarScanner.MSBuild.PreProcessor.Test
 {
     internal class MockObjectFactory : IPreprocessorObjectFactory
     {
-        private readonly ISonarQubeServer server;
-        private readonly IAnalyzerProvider analyzerProvider;
-        private readonly ITargetsInstaller targetsInstaller;
+        public TestLogger Logger { get; } = new();
+        public MockSonarQubeServer Server { get; } = new();
+        public Mock<ITargetsInstaller> TargetsInstaller { get; } = new Mock<ITargetsInstaller>();
+        public MockRoslynAnalyzerProvider AnalyzerProvider { get; } = new() { SettingsToReturn = new AnalyzerSettings { RulesetPath = "c:\\xxx.ruleset" } };
```

Comment:
This can be private. Maybe we should make it a field.

Reply:
I would follow the same pattern for everyone. They are all meant to be mocked and customized.